userspace/libsinsp/container_engine/docker_common.cpp

```diff
@@ -333,6 +339,9 @@ bool docker_async_source::parse_docker(std::string &container_id, sinsp_containe
 							container->m_imagedigest = repodigest.substr(repodigest.find("@")+1);
 							break;
 						}
+						if(prevdigest == "") {
```

Comment:
This is all hard to follow but I guess the idea is to return an imagedigest even if the image is not tagged in a repository. We probably want to prefer the digest tagged with the repo (as in current code), so we could do something like:

```
for repodigest in repodigests:
    if m_imagerepo in repodigest or m_imagedigest.empty():
        m_imagedigest = substr(...)
```

instead of prevdigest/currentdigest/foundsingledigest (since we loop over all the digests anyway)